sphinx/builders/html.py

```diff
@@ -746,79 +746,85 @@ def to_relpath(f):
+                copy_asset(entry, path.join(self.outdir, '_static'), excluded,
                            context=ctx, renderer=self.templates)
-        # then, copy over all user-supplied static files
-        excluded = Matcher(self.config.exclude_patterns + ["**/.*"])
-        for static_path in self.config.html_static_path:
-            entry = path.join(self.confdir, static_path)
-            if not path.exists(entry):
-                logger.warning('html_static_path entry %r does not exist', entry)
-                continue
-            copy_asset(entry, path.join(self.outdir, '_static'), excluded,
-                       context=ctx, renderer=self.templates)
-        # copy logo and favicon files if not already in static path
-        if self.config.html_logo:
-            logobase = path.basename(self.config.html_logo)
-            logotarget = path.join(self.outdir, '_static', logobase)
-            if not path.isfile(path.join(self.confdir, self.config.html_logo)):
-                logger.warning('logo file %r does not exist', self.config.html_logo)
-            elif not path.isfile(logotarget):
-                copyfile(path.join(self.confdir, self.config.html_logo),
-                         logotarget)
-        if self.config.html_favicon:
-            iconbase = path.basename(self.config.html_favicon)
-            icontarget = path.join(self.outdir, '_static', iconbase)
-            if not path.isfile(path.join(self.confdir, self.config.html_favicon)):
-                logger.warning('favicon file %r does not exist', self.config.html_favicon)
-            elif not path.isfile(icontarget):
-                copyfile(path.join(self.confdir, self.config.html_favicon),
-                         icontarget)
-        logger.info('done')
+            # copy logo and favicon files if not already in static path
+            if self.config.html_logo:
+                logobase = path.basename(self.config.html_logo)
+                logotarget = path.join(self.outdir, '_static', logobase)
+                if not path.isfile(path.join(self.confdir, self.config.html_logo)):
+                    logger.warning('logo file %r does not exist', self.config.html_logo)
+                elif not path.isfile(logotarget):
+                    copyfile(path.join(self.confdir, self.config.html_logo),
+                             logotarget)
+            if self.config.html_favicon:
+                iconbase = path.basename(self.config.html_favicon)
+                icontarget = path.join(self.outdir, '_static', iconbase)
+                if not path.isfile(path.join(self.confdir, self.config.html_favicon)):
+                    logger.warning('favicon file %r does not exist', self.config.html_favicon)
+                elif not path.isfile(icontarget):
+                    copyfile(path.join(self.confdir, self.config.html_favicon),
+                             icontarget)
+            logger.info('done')
+        except Exception as err:
```

Comment:
If we do not want to catch unexpected exceptions such as NameError that is caused by programming error, it would be nice to catch `OSError` instead of `Exception`.
https://docs.python.org/3.7/library/exceptions.html#exception-hierarchy

Reply:
In py2, `IOError` is independent exception class. So it is better to catch `EnvironmentError` instead. It is common parent class of `IOError` and `OSError` in py2. And it is an alias of `OSError` in py3.

Reply:
done

Reply:
Sounds good!